src/aws-cpp-sdk-core/include/aws/core/utils/Array.h

```diff
@@ -220,6 +262,29 @@ namespace Aws
             CryptoBuffer(CryptoBuffer&& other) : ByteBuffer(std::move(other)) {}
             CryptoBuffer& operator=(const CryptoBuffer&) = default;
             CryptoBuffer& operator=(CryptoBuffer&& other) { ByteBuffer::operator=(std::move(other)); return *this; }
+
+            CryptoBuffer(Crt::ByteBuf&& other) noexcept : ByteBuffer(
+                other.len,
+                other.len,
+                nullptr)
+            {
+                m_data.reset(other.buffer);
+                other.capacity = 0;
+                other.len = 0;
+                other.allocator = nullptr;
+            }
+
+            CryptoBuffer& operator=(Crt::ByteBuf&& other) noexcept
+            {
+                m_capacity = other.len;
```

Comment:
add a check that other != parent of this
(i.e. `static_cast<Crt::ByteBuf*>(this) != &other`) or smth like this
otherwise you risk erasing the parent

Reply:
`CryptoBuffer != Crt::ByteBuf`  we actually _want_ to delete the parent unconditionally as we are taking ownership of the underlying buffer from the bytebuf. since this is a move operation `other` is invalid after this operation and `CryptoBuffer` should always delete parent, and it cannot move from self as they are different types.